packages/suite/src/components/wallet/WalletLayout/components/CoinjoinAccountDiscovery/useAccountLoadingProgress.ts

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable no-nested-ternary */
+
+import { useEffect, useReducer } from 'react';
+import { ScanProgressInfo } from '@trezor/coinjoin/lib/types/backend';
+
+import { CoinjoinService } from 'src/services/coinjoin';
+import { selectSelectedAccount } from 'src/reducers/wallet/selectedAccountReducer';
+import { useSelector } from 'src/hooks/suite/useSelector';
+import { getAccountProgressHandle } from 'src/utils/wallet/coinjoinUtils';
+import { TranslationKey } from 'src/components/suite/Translation';
+
+const INIT_THRESHOLD = 0.05;
+const MEMPOOL_THRESHOLD = 0.85;
+
+type ProgressInfoReducerState = {
+    messageId?: TranslationKey;
+    progress: number;
+    stage?: 'block' | 'mempool';
+    outOf?: { current: number; total: number };
+};
+
+const progressInfoReducer = (
+    state: ProgressInfoReducerState,
+    info: ScanProgressInfo,
+): ProgressInfoReducerState => {
+    const outOf = !info.progress
```

Comment:
nit: the ternaries are very hard to read >_<

Reply:
Luckily, I have one more fixup commit which makes them even worse. :smiling_imp:  I'll add another fixup with if/else instead, but I don't like it either.